src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash64.State.cs

```diff
@@ -97,6 +97,7 @@ private static ulong ApplyRound(ulong acc, ulong lane)
                 return acc;
             }
 
+            [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Please add a comment about why this is here.

And as you noted it's temporary, is there an issue associated with removing it?

Reply:
The general issue is https://github.com/dotnet/runtime/issues/85531 we plan to make inliner a bit more advanced here and fix the budget calculation but it's unlikely to make it to net8.0 😞 I'll add this example to the issue

Reply:
Added a comment